### PR TITLE
🏗 🐛 Adjust Sauce Labs timeouts to prevent disconnections

### LINKED ISSUE #14848: Karma tests disconnect when running tests on Safari
When running the tests on Travis for my PR #14679, I'm getting: Test disconnected because Safari 11.0.1 is not generating any test output for 2 minutes. https://travis-ci.org/ampproject/amphtml/jobs/369330113#L725 Note that it eventually reports a success in L728.

Have tried running it several times and the problems seems to persist.

### LINKED ISSUE #16135: Build failures related to Sauce Labs timeouts / crashes / disconnects
Seeing integration test builds fail with no specific test failures due to timeout and disconnects from SauceLabs. 

Sample builds: 
https://travis-ci.org/ampproject/amphtml/jobs/393858309

Relevant stacktraces: 
```
Chrome 67.0.3396 (Windows 7 0.0.0) ERROR
  Disconnected, because no message in 240000 ms.
[22:49:29] Chrome 67.0.3396 (Windows 7 0.0.0): Executed 180 of 401 (Skipped 120) SUCCESS
18 06 2018 22:49:29.374:ERROR [reporter.sauce]: ✖ Test Disconnected
18 06 2018 22:50:49.639:ERROR [launcher.sauce]: Heartbeat to chrome latest failed
  [title()] Not JSON response
  ERROR The test with session id a711cbf73ece4b3184b087936d446c6e has already finished, and can't receive further commands.
You can learn more at https://saucelabs.com/jobs/a711cbf73ece4b3184b087936d446c6e
For help, please check https://wiki.saucelabs.com/display/DOCS/Common+Error+Messages
Chrome 67.0.3396 (Windows 7 0.0.0) ERROR
  Disconnected, because no message in 240000 ms.
[22:54:54] Chrome 67.0.3396 (Windows 7 0.0.0): Executed 180 of 401 (Skipped 120) SUCCESS
18 06 2018 22:54:54.405:ERROR [reporter.sauce]: ✖ Test Disconnected
```

### PR DESCRIPTION
**Pull request by rsimha**
Based on the documentation at https://wiki.saucelabs.com/display/DOCS/Test+Configuration+Options#TestConfigurationOptions-Timeouts

Fixes #16135 
Fixes #14848

### CODE REVIEW

File: build-system/tasks/karma.conf.js [hunk 1]
```diff
@@ -143,50 +149,77 @@ module.exports = {
       base: 'SauceLabs',
       browserName: 'chrome',
       version: 'latest',
+      maxDuration: SAUCE_MAX_DURATION_SECS,
+      commandTimeout: SAUCE_COMMAND_TIMEOUT_SECS,
+      idleTimeout: SAUCE_IDLE_TIMEOUT_SECS,
```
Comment (choumx) [inline build-system/tasks/karma.conf.js]: You can define shared properties once and then use `Object.assign` to copy them across each SL config.

```js
const SAUCE_CONFIG_TIMEOUT = {
  maxDuration: SAUCE_MAX_DURATION_SECS,
  commandTimeout: SAUCE_COMMAND_TIMEOUT_SECS,
  idleTimeout: SAUCE_IDLE_TIMEOUT_SECS,
};

...

SL_Chrome_latest: Object.assign({
  base: 'SauceLabs',
  browserName: 'chrome',
  version: 'latest',
}, SAUCE_TIMEOUT_CONFIG),
```

Comment (rsimha) [reply build-system/tasks/karma.conf.js]: Cool! Done.

File: build-system/tasks/karma.conf.js [hunk 2]
```diff
@@ -139,55 +147,55 @@ module.exports = {
...
-    SL_Chrome_45: {
+    }, SAUCE_TIMEOUT_CONFIG),
+    SL_Chrome_45: Object.assign({
       base: 'SauceLabs',
       browserName: 'chrome',
       version: '45',
-    },
-    SL_Android_latest: {
+    }, SAUCE_TIMEOUT_CONFIG),
+    SL_Android_latest: Object.assign({
```
Comment (cathyxz) [inline build-system/tasks/karma.conf.js]: Somewhat relatedly. I've noticed that most of our timeouts happen with Chrome on Android (the one that says Mobile Webview) and Chrome on Windows. I don't know if it's the case that those should be longer, or are we just more broken on said platforms.

Comment (rsimha) [reply build-system/tasks/karma.conf.js]: It's possible there's more than one thing at play. Let's merge this and observe the results.

**Pull request merged**